### PR TITLE
331

### PR DESCRIPTION
**Pull request by JasonGroven**
Issues getting 'formInputConfig' object from app.component to act as config object in form-input.ts, so I instantiate a fresh BossyFormInputConfig("text") in the form input class itself.  This will need to be changed.  Otherwise, this seems to render a text box in the sandbox app using the expected syntax.

### CODE REVIEW

File: src/components/form-input.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+})
+export class BossyFormInput {
+	public config: any;
+	@Input() options = new BossyFormInputConfig("text");
+    constructor(){
+       if(this.options.type === "text")
+       {
+			this.output();
+       }
+        console.log('option-type: ' + this.options.type);
```
Comment (glawson) [inline src/components/form-input.ts]: Remove the console.log

File: src/components/form-input.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+    ngOnInit() {
+      console.log('ngOnInit: option-type: ' + this.options.type);
```
Comment (glawson) [inline src/components/form-input.ts]: Remove the console.log

File: src/components/form-input.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+			this.output();
+       }
+        console.log('option-type: ' + this.options.type);
+    }
+    ngOnInit() {
+      console.log('ngOnInit: option-type: ' + this.options.type); 
+    }
+    output()
+    {
+		console.log("Component in BossyFormInput is outputting.");
```
Comment (glawson) [inline src/components/form-input.ts]: Remove the console.log

File: src/components/form.ts [hunk 4]
```diff
@@ -13,7 +13,7 @@ declare const module: any;
 export class BossyForm {
     configKeys: Array<string>;
     configForm: FormGroup;
-    @Input() config: any;
```
Comment (glawson) [inline src/components/form.ts]: Why this change?

File: src/templates/form-input.html [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<div class="bossy-form-input">
+   bossy-form-input for text:
+  <input type={{this.options.type}}/>
+  {{this.options.type}}
```
Comment (JasonGroven) [inline src/templates/form-input.html]: (This is just test output to make sure that the 'type' of options is correct.  Not necessary for function, in hindsight should be removed)

File: src/components/form.ts (hunk 4 above)
Comment (JasonGroven) [inline src/components/form.ts]: This could be causing the issue that I was having getting the config to pass into the form-input class.  I'll take a look at this as soon as I get home tonight.

File: src/templates/form-input.html [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<div class="bossy-form-input">
+   bossy-form-input for text:
```
Comment (tabersays) [inline src/templates/form-input.html]: The form inputs text should include extranious text

File: src/components/calendar.ts [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import {Component, OnInit} from '@angular/core';
+import {Component, OnInit, Input} from '@angular/core';
```
Comment (tabersays) [inline src/components/calendar.ts]: We don't want the input in the calander

File: src/templates/form-input.html [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+  <input type={{this.config.type}} />
```
Comment (tabersays) [inline src/templates/form-input.html]: May consider ng-if statement incase we have types that dont directly map to input types

Comment (tabersays) [inline src/templates/form-input.html]: with a default case if posible
